### DISCUSSION TITLE
💬 Roadmap Update July 2024

### DISCUSSION THREAD

**Discussion by chrisgurney** [chrisgurney/obsidian-note-toolbar#85] [Announcements]
_Going forward I'll be updating the [Roadmap](https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Roadmap) page, so be sure to check back there for updates!_

Hello all! 

Firstly, I wanted to mention why I haven't released as much as of late: I've been looking for a job on and off, while working on personal projects (such as this one), since November. My focus has been on the job search lately and on top of that I have an upcoming move to prepare for. This doesn't leave me with a lot of project time, unfortunately.

At the very least then, I wanted to share my current thinking on the next few versions I'll be working on, based on your great ideas (thank you!).

## v1.8 - [Released](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.8)

Includes usability improvements to settings (including drag-and-drop) as it inevitably has started to get more complex.

## v1.9 (current version) - [Released](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.9.1)

_**Update July 2024:** With the goal of more frequent releases, I've opted to release v1.9 to include only Item Menus, and will work on Item Groups in v1.10. This means new mapping types will be in v1.11 or beyond, depending on priorities at that time._

There's been a lot of requests for sub-toolbars, toolbar inheritance, and menus -- thanks again for your ideas -- so I'll be focussing on adding a new type: _Item Menu_

Add a _Menu_ type item to the end of Toolbar 2 instead, that's a reference to Toolbar 1:

- Toolbar 2 will have these items: `D` `E` `F`
- When `F` is clicked, it opens a drop-down menu containing: `A` `B` `C`

## v1.10 

This one will add the type: _Item Group_ 

As an example:

- Toolbar 1 has these items: `A` `B` `C`
- Toolbar 2 has these items: `D` `E`

Let's say I add another _Item Group_ type item to the end of Toolbar 2 that's just a reference to Toolbar 1's items:

- Now Toolbar 2 has these items: `D` `E` `A` `B` `C`
- Toolbar 2 will continue to be styled and positioned per Toolbar 2's settings.

## v1.11 or later

The current idea, just beyond the above theme, is to extend toolbar (folder) mappings to include a couple more types:

- `Tags`, so that toolbars can apply to types of notes, regardless of folder.
  - Note: This can sort of be accomplished today by changing the `notetoolbar` property to `tags` in settings, assuming you name your toolbars according to your tags.
- `File Patterns`, so that those who name their folders based on dates for example (e.g., daily notes are in `2024/06`) will be able to map a toolbar to a regular expression pattern.

_As mentioned, going forward I'll be updating the [Roadmap](https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Roadmap) page, so be sure to check back there for updates!_

**Comment by FelipeRearden** [comment]
> File Patterns, so that those who name their folders based on dates for example (e.g., daily notes are in 2024/06) will be able to map a toolbar to a regular expression pattern.

@chrisgurney , The source code from the Path Title Plugin could save you time when developing this feature 🙏

https://github.com/jdeal/obsidian-path-title-plugin#obsidian-path-title
